x-pack/plugins/alerting/server/rules_client/rules_client.ts

```diff
@@ -631,6 +647,56 @@ export class RulesClient {
     });
   }
 
+  public async getExecutionLogForRule({
+    id,
+    dateStart,
+    dateEnd,
+    filter,
+    page,
+    perPage,
+    sort,
+  }: GetExecutionLogByIdParams): Promise<IExecutionLogResult> {
+    this.logger.debug(`getExecutionLogForRule(): getting execution log for rule ${id}`);
+    const rule = (await this.get({ id, includeLegacyId: true })) as SanitizedRuleWithLegacyId;
+
+    // Make sure user has access to this rule
+    await this.authorization.ensureAuthorized({
+      ruleTypeId: rule.alertTypeId,
+      consumer: rule.consumer,
+      operation: ReadOperations.GetExecutionLog,
+      entity: AlertingAuthorizationEntity.Rule,
+    });
+
+    // default duration of instance summary is 60 * rule interval
+    const dateNow = new Date();
+    const parsedDateStart = parseDate(dateStart, 'dateStart', dateNow);
+    const parsedDateEnd = parseDate(dateEnd, 'dateEnd', dateNow);
+
+    const eventLogClient = await this.getEventLogClient();
+
+    const results = await eventLogClient.aggregateEventsBySavedObjectIds(
+      'alert',
+      [id],
+      {
+        start: parsedDateStart.toISOString(),
+        end: parsedDateEnd.toISOString(),
+        filter,
+        aggs: getExecutionLogAggregation({
+          // determine the number of executions to request
+          // based on the date interval and the rule schedule interval
+          // this value is capped by the max number of terms allowed in a term aggregation
+          numExecutions: getNumExecutions(parsedDateStart, parsedDateEnd, rule.schedule.interval),
```

Comment:
Not of concern now, but once ad-hoc rule runs are introduced we may have to rework this logic to include some buffer for those additional executions.

Reply:
Another corner case here would be if the rule interval has been changed at all during the query daterange. Should be negligible unless it was a larger interval change (1min to say 10min), but still another way to miss events. FWIW I ended up foregoing this check and am just capping to the defined `DEFAULT_MAX_BUCKETS_LIMIT` on our end which ended up being `1000` for now.

Reply:
I like that simplification! Done here: https://github.com/elastic/kibana/pull/127339/commits/8ca934de33f91af9b181b34fac5f053313c5d471